cooltools/cli/compute_saddle.py

```diff
@@ -0,0 +1,515 @@
+    digitized, binedges = saddle.digitize_track(
+                                    bins = n_bins,
+                                    get_track = track_fetcher,
+                                    get_mask  = track_mask_fetcher,
+                                    chromosomes = track_chroms,
+                                    prange = prange,
+                                    by_percentile = by_percentile)
+    # digitized fetcher, yielding per chrom slice:
+    digitized_fetcher = lambda chrom: digitized[chrom]
+
+
+    # Aggregate contacts (implicit contact_type dependency):
+    sum_, count = saddle.make_saddle(
+        get_matrix = obsexp_func, 
+        get_digitized = digitized_fetcher,
+        chromosomes = track_chroms,
+        contact_type = contact_type,
+        verbose = verbose)
+    # actual saddleplot data:
+    saddledata = sum_/count
+
+    ##############################
+    # OUTPUT AND PLOTTING:
+    ##############################
+
+    # # output to stdout,
+    # # just like in diamond_insulation:
+    if output is not None:
+        # output saddledata (square matrix):
+        pd.DataFrame(saddledata).to_csv(output+".saddledata.tsv",
+                                        sep='\t',
+                                        index=False,
+                                        header=False,
+                                        na_rep='nan')
+        # output digitized track:
+        # we should be able to reuse 'track' DataFrame
+        # more than we're doing now. Are we trusing it ?
+        var_name = "chrom"
+        value_name = "digitized."+track_name
+        pd.melt(
+            pd.DataFrame.from_dict(digitized,orient="index").transpose(),
+            var_name=var_name,
+            value_name=value_name) \
+                .dropna() \
+                .reset_index(drop=True) \
+                .astype({var_name: np.str, value_name: np.int64}) \
+                .to_csv(output+".digitized.tsv",
+                    sep="\t",
+                    index=False,
+                    header=True,
+                    na_rep='nan')
+        # output binedges:
+        pd.Series(binedges, name="binedges."+track_name).\
+            to_csv(output+".binedges.tsv",
+                sep="\t",
+                index=False,
+                header=True,
+                na_rep='nan')
+    else:
+        print(pd.DataFrame(saddledata).to_csv(
+                                        sep='\t',
+                                        index=False,
+                                        header=False,
+                                        na_rep='nan'))
+    
+
+
+
+    # ###########################
+    # do the plotting, if requested ...
+    # from cooler show cli:
+    # if savefig or something like that:
+    if savefig is not None:
+        try:
+            import matplotlib as mpl
+            # savefig only for now:
+            mpl.use('Agg')
```

Comment:
Yes! This should be the default, especially if running on headless computers.

Reply:
yeah, it was simply copy/pasted from your `cooler_show` implementation: https://github.com/mirnylab/cooler/blob/master/cooler/cli/show.py